dblookupext/miniblocksHandler.go

```diff
@@ -280,3 +263,106 @@ func (mh *miniblocksHandler) removeMiniblockMetadata(
+	miniblockMetadataNextEpoch, err := mh.loadExistingMiniblocksMetadata(miniblockHash, epoch+1)
+	if err != nil {
+		// no, return what we found
+		return miniblockMetadataFirstEpoch, nil
+	}
+	miniblockMetadataFirstEpoch.MiniblocksInfo = append(miniblockMetadataFirstEpoch.MiniblocksInfo, miniblockMetadataNextEpoch.MiniblocksInfo...)
+
+	return miniblockMetadataFirstEpoch, nil
+}
+
+func (mh *miniblocksHandler) getMiniblockMetadataByMiniblockHash(miniblockHash []byte) ([]*MiniblockMetadata, error) {
+	miniblockMetadata, err := mh.retrieveMiniblockMetadata(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return convertMiniblockMetadata(miniblockMetadata), nil
+}
+
+func (mh *miniblocksHandler) retrieveMiniblockMetadata(miniblockHash []byte) (*MiniblockMetadataV2, error) {
+	epoch, err := mh.epochIndex.getEpochByHash(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	return mh.loadAllExistingMiniblocksMetadata(miniblockHash, epoch)
+}
+
+func (mh *miniblocksHandler) getMiniblockMetadataByTxHash(txHash []byte) (*MiniblockMetadata, error) {
+	miniblockHash, err := mh.miniblockHashByTxHashIndexStorer.Get(txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	miniblockMetadata, err := mh.retrieveMiniblockMetadata(miniblockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mbInfo := range miniblockMetadata.MiniblocksInfo {
+		if findTxHashInMiniblockMetadataOnBlock(mbInfo, txHash) {
+			miniblockMetadata.MiniblocksInfo = []*MiniblockMetadataOnBlock{mbInfo}
+			mbData := convertMiniblockMetadata(miniblockMetadata)
+			return mbData[0], nil
+		}
+	}
+
+	return nil, fmt.Errorf("programming error, %w for txhash %x and mb hash %x",
+		ErrNotFoundInStorage, txHash, miniblockHash)
+}
+
+func convertMiniblockMetadata(miniblockMetadata *MiniblockMetadataV2) []*MiniblockMetadata {
+	result := make([]*MiniblockMetadata, 0, len(miniblockMetadata.MiniblocksInfo))
+	for _, mbInfo := range miniblockMetadata.MiniblocksInfo {
+		mbv1 := &MiniblockMetadata{
+			SourceShardID:                     miniblockMetadata.SourceShardID,
+			DestinationShardID:                miniblockMetadata.DestinationShardID,
+			Round:                             mbInfo.Round,
+			HeaderNonce:                       mbInfo.HeaderNonce,
+			HeaderHash:                        mbInfo.HeaderHash,
+			MiniblockHash:                     miniblockMetadata.MiniblockHash,
+			Epoch:                             mbInfo.Epoch,
+			NotarizedAtSourceInMetaNonce:      mbInfo.NotarizedAtSourceInMetaNonce,
+			NotarizedAtDestinationInMetaNonce: mbInfo.NotarizedAtDestinationInMetaNonce,
+			NotarizedAtSourceInMetaHash:       mbInfo.NotarizedAtSourceInMetaHash,
+			NotarizedAtDestinationInMetaHash:  mbInfo.NotarizedAtDestinationInMetaHash,
+			Type:                              miniblockMetadata.Type,
+		}
+
+		result = append(result, mbv1)
+	}
+
+	return result
+}
+
+func findTxHashInMiniblockMetadataOnBlock(mbInfo *MiniblockMetadataOnBlock, txHash []byte) bool {
```

Comment:
`find` makes us think that an item is returned. Alternative: `existsTxHashInMiniblockMetadataOnBlock`.